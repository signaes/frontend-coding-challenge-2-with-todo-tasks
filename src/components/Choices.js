import React, {PureComponent} from 'react';

export class Choices extends PureComponent {
  render () {
    return (
      <table>
        <thead>
          <tr>
            <th>Choices</th>
            <th>Votes</th>
            <th>Percentage</th>
            {this.props.onVote &&
              <th></th>
            }
          </tr>
        </thead>
        <tbody>
          {this.props.choices.map(({choice, url, votes}) => {
            const percentage = Math.round(votes / this.props.totalVotes * 100);
            return (
              <tr key={url}>
                <td>{choice}</td>
                <td>{votes || 0}</td>
                <td>{percentage}%</td>
                {this.props.onVote &&
                  <td>
                    <button
                      disabled={this.props.fetching}
                      onClick={() => this.props.onVote(url)}>
                      Vote
                    </button>
                  </td>
                }
              </tr>
            )
          })}
        </tbody>
      </table>
    )
  }
}

export default Choices;
