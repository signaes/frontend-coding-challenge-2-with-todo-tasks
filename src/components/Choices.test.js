import React from 'react';
import {shallow} from 'enzyme';

import Choices from './choices';

const choices = [
  {url: 'url1', choice: 'a', votes: '1'},
  {url: 'url2', choice: 'b', votes: '10'},
  {url: 'url3', choice: 'c', votes: '100'},
];

it('matches the snapshot', () => {
  const wrapper = shallow(<Choices choices={choices}></Choices>);
  expect(wrapper).toMatchSnapshot();
})
