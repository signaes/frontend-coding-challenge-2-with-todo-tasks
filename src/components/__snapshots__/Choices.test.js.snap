// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Choices
      </th>
      <th>
        Votes
      </th>
      <th>
        Percentage
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="a-url1"
    >
      <td>
        a
      </td>
      <td>
        1
      </td>
      <td>
        NaN
        %
      </td>
    </tr>
    <tr
      key="b-url2"
    >
      <td>
        b
      </td>
      <td>
        10
      </td>
      <td>
        NaN
        %
      </td>
    </tr>
    <tr
      key="c-url3"
    >
      <td>
        c
      </td>
      <td>
        100
      </td>
      <td>
        NaN
        %
      </td>
    </tr>
  </tbody>
</table>
`;
